### DISCUSSION TITLE
Cache-image was used, but it still took a long time to build the vue project

### DISCUSSION THREAD

**Discussion by Sunny-001** [paketo-buildpacks/feedback#49] [General]
I used the following command to build the vue project, and when I built it on another server with the same command, it still took longer than not using cache-image

**command:**
pack build registry.cn-hangzhou.aliyuncs.com/kpack/ruoyi-ui:2.0.0 \
  --path ./ruoyi-ui/ \
  --builder registry.cn-hangzhou.aliyuncs.com/kpack/paketo-buildpacks-builder:base \
  --env "BP_NODE_RUN_SCRIPTS=build" \
  --env "NODE_ENV=development" \
  --pull-policy if-not-present \
  --cache-image registry.cn-hangzhou.aliyuncs.com/kpack/nodejs-cache:latest \
  --publish

**Comment by dmikusa** [comment]
The first time you run `pack build` for an application, which is identified by the image name you specify, it will have nothing in the cache. Thus it'll take about the same time it does for a clean build on a new machine or container, because that is exactly what `pack build` is doing it's performing a clean build. After that, when you rebuild, you should see it run quite a bit faster, depending on what's changed, as it can cache a lot of the work that was done on the first run.

The key to seeing good caching results is to not change the image name for your application. The cache is scoped to the image name, so if you change the image name, you're getting a new build with no cache.

If you have further questions about whether something is or isn't cached, please post the output of your builds and indicate if it's a first build or subsequent build.
